pkg/kubelet/kubelet_pods.go

```diff
@@ -453,13 +469,13 @@ func (kl *Kubelet) GenerateRunContainerOptions(pod *v1.Pod, container *v1.Contai
 		opts.Devices = append(opts.Devices, blkVolumes...)
 	}
 
-	mounts, cleanupAction, err := makeMounts(pod, kl.getPodDir(pod.UID), container, hostname, hostDomainName, podIP, volumes, kl.mounter)
+	mounts, cleanupAction, err := makeMounts(pod, kl.getPodDir(pod.UID), container, hostname, hostDomainName, podIP, volumes, kl.mounter, expandEnvs)
 	if err != nil {
 		return nil, cleanupAction, err
 	}
 	opts.Mounts = append(opts.Mounts, mounts...)
 
-	envs, err := kl.makeEnvironmentVariables(pod, container, podIP)
+	envs, err := kl.GenerateRunContainerEnvs(pod, container, podIP)
```

Comment:
Can this be moved before makeMounts? Then you don't need to pass expandEnvs into GenerateRunContainerOptions

Reply:
@msau42 Yes, that looks a good idea as it now looks like it flows a bit better. Have moved the generate to just before the makeMounts.